pom.xml

```diff
@@ -435,9 +429,35 @@
                             </execution>
                         </executions>
                     </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>central-deploy</id>
+            <build>
+                <plugins>
                     <plugin>
-                        <groupId>org.eluder.coveralls</groupId>
-                        <artifactId>coveralls-maven-plugin</artifactId>
+                        <groupId>org.jacoco</groupId>
+                        <artifactId>jacoco-maven-plugin</artifactId>
+                        <executions>
+                            <!-- Prepares the property pointing to the JaCoCo runtime agent which is passed as
+                                 VM argument when Maven the Surefire plugin is executed. -->
+                            <execution>
+                                <id>pre-unit-test</id>
+                                <goals>
+                                    <goal>prepare-agent</goal>
+                                </goals>
+                            </execution>
+                            <!-- Ensures that the code coverage report for unit tests is created after unit tests have
+                                 been run. -->
+                            <execution>
+                                <id>post-unit-test</id>
+                                <phase>test</phase>
+                                <goals>
+                                    <goal>report</goal>
```

Comment:
Why do you bind the report goal to the **test** phase? Being executed in the same phase as the tests means it could theoretically be executed before the tests? This would also miss the results of integration test phase. The default would be **verify** which sounds more reasonable to me.

Reply:
By default it binds to the verify phase. Which is better, since that covers both unit and integration tests (should we ever have them).